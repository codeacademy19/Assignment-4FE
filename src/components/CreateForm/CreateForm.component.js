import React, { Component } from 'react';
import {
  SafeAreaView, ImageBackground, Text, View, Image, TouchableOpacity, TextInput, ScrollView, Button,
} from 'react-native';
import PropTypes from 'prop-types';
import styles from './CreateForm.css';


class FormCard extends Component {
    state={
      Attributes: '',
      name: '',
      text: '',
      rows: [],
    }


    handleChangeText = (text) => {
      this.setState({ text });
    };

      handleSubmitEditing = () => {
        if (!this.state.text) return;

        this.props.onSubmit(this.state.text);
        this.setState({ text: '' });
      };

      handleOnPress=() => {
        this.setState((state) => {
          state.rows.append(<TextInput
            value={this.state.text}
            autoCorrect={false}
            placeholder="Field Name"
            placeholderTextColor="black"
            underlineColorAndroid="transparent"
            style={{
              width: '85%',
              height: 50,
              marginTop: 40,
              marginHorizontal: 20,
              paddingHorizontal: 10,
              alignSelf: 'flex-start',
              borderBottomWidth: 2,
              borderBottomColor: 'grey',
              fontSize: 18,
            }}
            onChangeText={this.handleChangeText}
            onSubmitEditing={this.handleSubmitEditing}
          />);
        });
        console.log(this.state);
      }

      render() {
        return (
          <SafeAreaView style={styles.container}>
            <View style={styles.header}>
              <Text style={styles.headerText}>Create Form</Text>
            </View>
            <ScrollView
              contentContainerStyle={{
                flexGrow: 1,
                justifyContent: 'flex-start',
                alignItems: 'center',
              }}
              style={{ width: '100%', backgroundColor: 'white' }}
            >


              <TextInput
                value={this.state.text}
                autoCorrect={false}
                placeholder="Form Name"
                placeholderTextColor="black"
                underlineColorAndroid="transparent"
                style={{
                  width: '85%',
                  height: 50,
                  marginTop: 40,
                  marginHorizontal: 20,
                  paddingHorizontal: 10,
                  alignSelf: 'flex-start',
                  borderBottomWidth: 2,
                  borderBottomColor: 'grey',
                  fontSize: 18,
                }}
                onChangeText={this.handleChangeText}
                onSubmitEditing={this.handleSubmitEditing}
              />
              <TouchableOpacity
                style={styles.button}
                onPress={() => this.handleOnPress}
              >
                <Text style={styles.buttonText}>
                  {' '}
                  {'ADD FIELD'}
                  {' '}
                </Text>
                {this.state.rows}
              </TouchableOpacity>
            </ScrollView>


          </SafeAreaView>


        );
      }
}

// FormCard.defaultProps = {
//   name: 'default',
//   time: '15/01/01',
// };

FormCard.propTypes = {
  name: PropTypes.string.isRequired,
  time: PropTypes.string.isRequired,
};

export default FormCard;
