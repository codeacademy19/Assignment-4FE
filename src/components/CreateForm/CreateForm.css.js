import { StyleSheet } from 'react-native';
import { FONT_HEADER, FORM_NAME } from '../../constants';

export default StyleSheet.create({
  scroll: {
    width: '100%',
    height: '100%',

  },
  header: {
    backgroundColor: '#1d61e0',
    justifyContent: 'center',
    width: '100%',
    height: '10%',

  },
  headerText: {
    fontSize: FONT_HEADER,
    color: 'white',
    paddingLeft: 25,
  },
  container: {
    flex: 1,
    alignItems: 'center',
    backgroundColor: 'white',
    height: '100%',
    width: '100%',
  },
  buttonText: {
    fontSize: FORM_NAME,
    color: 'white',
    paddingTop: 7,
  },
  button: {
    width: '35%',
    height: '7%',
    margin: 30,
    backgroundColor: '#1d61e0',
    alignItems: 'center',
    alignSelf: 'flex-end',
  },
  textInput: {
    width: 300,
    height: 50,
    marginTop: 40,
    backgroundColor: 'grey',
    marginHorizontal: 20,
    paddingHorizontal: 10,
    alignSelf: 'flex-start',
    borderWidth: 2,
  },
});
