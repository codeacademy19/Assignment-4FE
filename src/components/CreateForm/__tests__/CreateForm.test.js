import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import CreateForm from '../CreateForm.component';


describe('CardView component', () => {
  it('renders as expected', () => {
    const tree = renderer.create(
      <CreateForm />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
