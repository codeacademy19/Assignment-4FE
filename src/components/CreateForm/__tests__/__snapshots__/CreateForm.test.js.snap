// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardView component renders as expected 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "height": "100%",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#1d61e0",
        "height": "10%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 22,
          "paddingLeft": 25,
        }
      }
    >
      Create Form
    </Text>
  </View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "flexGrow": 1,
        "justifyContent": "flex-start",
      }
    }
    style={
      Object {
        "backgroundColor": "white",
        "width": "100%",
      }
    }
  >
    <View>
      <TextInput
        allowFontScaling={true}
        autoCorrect={false}
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Form Name"
        placeholderTextColor="black"
        style={
          Object {
            "alignSelf": "flex-start",
            "borderBottomColor": "grey",
            "borderBottomWidth": 2,
            "fontSize": 18,
            "height": 50,
            "marginHorizontal": 20,
            "marginTop": 40,
            "paddingHorizontal": 10,
            "width": "85%",
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "flex-end",
            "backgroundColor": "#1d61e0",
            "height": "7%",
            "margin": 30,
            "opacity": 1,
            "width": "35%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 21,
              "paddingTop": 7,
            }
          }
        >
           
          ADD FIELD
           
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
